recipes/dev/lora_finetune_fsdp2.py

```diff
@@ -314,6 +316,8 @@ def _setup_model(
         for m in reversed(list(model.modules())):
             if isinstance(m, nn.Linear) and m.weight.requires_grad:
                 fully_shard(m, **fsdp_kwargs)
+            if isinstance(m, DoRALinear):
```

Comment:
why do we treat DoRALinear separately here? why don't we treat it same as LoRALinear?

Reply:
I think it's necessary for proper initialization of the magnitude from the sharded LoRA A and B weights. Since otherwise they will be in different FSDP param groups so we would probably need to manually gather and then reshard the weights. You can doublecheck by commenting out L319-320 though to do the usual LoRA sharding, my guess is you will see some kind of DTensor error though.